### PR TITLE
Unique Email for Existing User Edit

### LINKED ISSUE #64: User Story 19, User Editing Profile Data must have unique Email address
```
As a registered user
When I attempt to edit my profile data
If I try to change my email address to one that belongs to another user
When I submit the form
Then I am returned to the profile edit page
And I see a flash message telling me that email address is already in use
```

DO THESE STORIES FIRST:
- User Story 17, User Profile Show Page
- User Story 18, User Can Edit their Profile Data

### Checklist:

- [x] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #253

child of #274

depends on #252

depends on #251

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the same logic from our User#create method to ensure that a User editing their profile is not able to change their Email Address to one already being used by another User.
This logic block also brings the Password confirmation as a bonus.

This PR also refactors the UsersController Email_confirmation method to use .include? instead. It also works for both the creation of a User, and for the updating of a User.
When Updating, you give it an argument of user_email, which is then removed from the pool of Existing Users. This allows a User to submit the SAME email while editing. Otherwise, that argument is ignored (as `nil`) and pulls all email addresses during creation.

Resolves #64

### CODE REVIEW

**Review by zachlp2016 — APPROVED** [review]
I see some refactoring on some helper methods.  Great job with that.  Also good looking test. Gouda job.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -33,19 +33,26 @@ def edit
...
+    if password_confirmation != true
+      flash.now[:notice] = "Those passwords don't match."
+      render :edit
+    elsif email_confirmation(@user.email) == true
+      flash.now[:notice] = "That email address is already taken."
+      render :edit
+    elsif @user.update!(strong_params)
+      flash[:notice] = "Your information has been updated!"
+      redirect_to profile_path
+    else
```
Comment (VinceCarollo) [inline app/controllers/users_controller.rb:46]: Solid sad path

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -33,19 +33,26 @@ def edit
...
-    user = current_user
-    user.update(strong_params)
-    flash[:notice] = "Your information has been updated!"
-
-    redirect_to profile_path
+    @user = current_user
+    if password_confirmation != true
+      flash.now[:notice] = "Those passwords don't match."
+      render :edit
+    elsif email_confirmation(@user.email) == true
```
Comment (VinceCarollo) [inline app/controllers/users_controller.rb:40]: We could refactor this to just if !password_confirmation

**Review by VinceCarollo — APPROVED** [review]
Solid logic. I like the way your brain is put together. 🧠
